### PR TITLE
feat: added auth and notes-card

### PR DESCRIPTION
**Pull request by xii-vi**
#### notes card

mobile (15)

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,4 +1,12 @@
 body{
-    
     font-family: 'Syne', sans-serif;
+}
+.sidebar-btn{
+    width: max-content;
+    color: black;
```
Comment (TejasShekar) [inline src/App.css:6]: use variables

File: src/App.css [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
+    margin-top: 1rem;
+}
+.note-page-layout{
+    display: grid;
+    grid-template-columns: 30% 60%;
```
Comment (TejasShekar) [inline src/App.css:11]: use grids from your CL

File: src/App.js [hunk 3]
```diff
@@ -2,15 +2,19 @@ import "./App.css";
 import { Navbar } from "./components/navbar/navbar";
 import { Homepage } from "./pages/homepage/homepage";
 import { Route, Routes} from "react-router-dom";
+import { Notes } from "./pages/note/note";
```
Comment (TejasShekar) [inline src/App.js:4]: create an index file and do named exports. it will shortens the path as well

File: src/index.js [hunk 4]
```diff
@@ -4,15 +4,21 @@ import "./index.css";
...
+import { AuthProvider } from "./context/authContext";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
+    <AuthProvider>
+    <NoteProvider>
     <BrowserRouter >
```
Comment (TejasShekar) [inline src/index.js:17]: router should be the on top of all context IG. Its always better to have library providers above all.

File: src/pages/signUp/signup.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import axios from "axios";
+import {useReducer} from "react";
+// import "../signUp/signup.css"
```
Comment (TejasShekar) [inline src/pages/signUp/signup.jsx:3]: remove unnecessary comments

File: src/reducer/authReducer.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        case "USER_LOGIN":
+            return {...state, userLogin:true}
+        case "USER_LOGOUT":
+            return {...state, userLogin:false, encodedToken:"", userData:""}
+        case "USER_DATA":
+            return {...state, userData:dispatch.payload}
+        case "USER_TOKEN":
+            return {...state, encodedToken:dispatch.payload}
+        default:
+            // state;
```
Comment (TejasShekar) [inline src/reducer/authReducer.jsx:13]: `return state`

File: src/reducer/authReducer.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+const loginReducer=(state, dispatch)=>{
+ switch(dispatch.type){
+    case "SET_USERNAME":  
+            return {...state, username:dispatch.payload}
+    case "SET_EMAIL":
+            return {...state, email:dispatch.payload}
+    case "SET_PASSWORD":
+            return {...state, password:dispatch.payload}
+        default:
+            // state;
```
Comment (TejasShekar) [inline src/reducer/authReducer.jsx:26]: `return state`

**Review by TejasShekar — APPROVED** [review]
suggested a few things. rest looks good to merge.

File: src/components/noteCard/notecard.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  )
+}
+const check = () => newNoteDispatch({type: "PIN_NOTE"});
+    return(
+        <form className="notes-form" onSubmit={(e)=>addToNoteHandler(e)}>
+          <div className="note-title flex">
+          <input
+          type="text"
+          placeholder="Title"
+          // value={title}
```
Comment (sharmayatendra) [inline src/components/noteCard/notecard.jsx:22]: remove commented code if  not needed in future

**Review by sharmayatendra — APPROVED** [review]
nothing major to change can look upon suggested changes by other reviewer also good to go

File: src/reducer/authReducer.jsx (hunk 7 above)
Comment (ckahub) [reply src/reducer/authReducer.jsx:26]: either remove default or un comment state

**Review by ckahub — COMMENTED** [review]
code looks neat nd clean
good naming of variable
code is very well readable

**Pull request merged**